src/main/scala/com/twitter/diffy/proxy/HttpDifferenceProxy.scala

```diff
@@ -34,8 +34,12 @@ trait HttpDifferenceProxy extends DifferenceProxy {
   override type Rep = HttpResponse
   override type Srv = HttpService
 
+
   override def serviceFactory(serverset: String, label: String) =
-    HttpService(Http.newClient(serverset, label).toService)
+    HttpService(
+      CloneHttpRequestFilter.apply.
+        andThen(RefineHttpHeadersByLabel(label)).
```

Comment:
Taking a step back, RefineHttpHeadersByLabel might be better written as two separate filters:

RewriteHeadersForLabel(label)
RemoveHeadersForLabels(labels)

and then composing them as:

```scala
CloneHttpRequestFilter.apply.
  andThen RewriteHttpHeadersByLabel(label)
  andThen RemoveHeadersForLabels(DifferenceProxy.AllLabels filter { !_.equalsIgnoreCase(label) })
```